internal/ui/processpane.go

```diff
@@ -35,38 +37,40 @@ func NewProcessPane(tApp *tview.Application, pr processrunner.ProcessRunner) *Pr
 	ansiWriter := tview.ANSIWriter(tv)
 
 	return &ProcessPane{
-		tApp:       tApp,
-		textView:   tv,
-		ansiWriter: ansiWriter,
-		pr:         pr,
-		Title:      pr.GetTitle(),
+		tApp:           tApp,
+		textView:       tv,
+		ansiWriter:     ansiWriter,
+		pr:             pr,
+		Title:          pr.GetTitle(),
+		TickerInterval: 250,
 	}
 }
 
 // StartScan starts scanning the stdout of the process and writes to the textView
 func (pp *ProcessPane) StartScan() {
-	// scan stdout and write using ansiWriter
 	go func() {
-		stdoutScanner := pp.pr.GetScanner()
-		for stdoutScanner.Scan() {
-			line := stdoutScanner.Text()
-			pp.tApp.QueueUpdateDraw(func() {
-				_, err := pp.ansiWriter.Write([]byte(line + "\n"))
-				pp.lineCount++
-				if err != nil {
-					fmt.Println("error writing to textView:", err)
-					panic(err)
-				}
-			})
-		}
-		if err := stdoutScanner.Err(); err != nil {
-			fmt.Println("error reading stdout:", err)
-			panic(err)
-		}
-		// FIXME - do i need to wait??
-		if err := pp.pr.Wait(); err != nil {
-			fmt.Println("error waiting for process:", err)
-			panic(err)
+		// initialize a ticker for periodic updates
+		ticker := time.NewTicker(pp.TickerInterval * time.Millisecond) // adjust the duration as needed
+		defer ticker.Stop()
+
+		var lastOutputSize int // tracks the last processed output size
+
+		for range ticker.C {
+			currentOutput := pp.pr.GetOutput() // get the current full output
+			currentSize := len(currentOutput)
+
+			if currentSize > lastOutputSize {
+				// new, unprocessed data.
+				newOutput := currentOutput[lastOutputSize:] // extract new data since last check
+				pp.tApp.QueueUpdateDraw(func() {
+					_, err := pp.ansiWriter.Write([]byte(newOutput))
+					pp.lineCount++
```

Comment:
Because of the addition of the ticker, this is now tracking number of writes and not the number of lines, which causes a bug with the '1' tail hotkey to only jump part way to the bottom of the logs instead of all the way.
This line needs to become something like `pp.lineCount += countNewlines([]byte(newOutput))` because there can now be several lines in each write if the ticker is set to a long time.